### DISCUSSION TITLE
AWS IoT Fleet Provisioning event template body and variables issue

### DISCUSSION THREAD

**Discussion by rsulzenbacher** [serverless/serverless#10872] [Q&A] [answered]
I'm trying to define the 'iotFleetProvisioning' event according to:
https://www.serverless.com/framework/docs/providers/aws/events/iot-fleet-provisioning
A template body is required and must be defined as object:
`templateBody: ${file(template.json)}`

My problem is, the template.json has for example following lines in it:
```
"Resource": [
  "arn:aws:iot:eu-west-1:1234567:client/${iot:Connection.Thing.ThingName}"
],
```

When I'm trying to deploy my service I get:
`Unrecognized configuration variable sources: "iot"`

In my case I wanna keep the content of template.json as it is and no variables should be replaced at all.

Any idea how I could do that?

**Comment by pgrzesik** [comment, marked as answer]
Hello @rsulzenbacher - sorry for not responding sooner, I've missed this discussion previously. It is possible to avoid attempt for variable resolution for specific values by escaping it e.g. like this:

```
provider:
  name: aws
  runtime: nodejs12.x
  lambdahashingversion: 20201221
  environment:
    val: \${iot:123}
```

During processing the backslash will be removed and the part with variable-like syntax will be left intact. Does that resolve your problem?

**Reply by rsulzenbacher** [reply]
Hi @pgrzesik! Thanks a lot, that helped!
